src/fparser/two/utils.py

```diff
@@ -380,9 +411,45 @@ def __new__(cls, string, parent_cls=None):
             errmsg = u"{0}: '{1}'".format(cls.__name__, string)
         raise NoMatchError(errmsg)
 
+    def get_root(self):
```

Comment:
I would change this to a `root` property to be consistent with parent and children, but would accept as is if you prefer it this way.

Reply:
I'm keeping it as get_root() as, as discussed, it could get costly to repeatedly call it if we're in a deep tree.